lib/rouge/lexers/javascript.rb

```diff
@@ -125,8 +129,13 @@ def self.builtins
           Array Boolean Date Error Function Math netscape
           Number Object Packages RegExp String sun decodeURI
           decodeURIComponent encodeURI encodeURIComponent
-          Error eval isFinite isNaN parseFloat parseInt document this
-          window
+          Error eval isFinite isNaN parseFloat parseInt
+          document window navigator self global
```

Comment:
In browsers, `self` is a global that points to the global object. Similar to `window`, but, unlike `window`, it also exists in web workers.

`global` is the global object in Node and, in a stage-3 proposal, should also become the global object in all environments.